src/rust/catnap/mod.rs

```diff
@@ -214,34 +155,17 @@ impl CatnapLibOS {
             }
         }
 
-        // Get a mutable reference to the queue table.
-        // The following call to expect() is safe because we checked before that the queue descriptor is valid.
-        let queue: &mut CatnapQueue = qtable
-            .get_mut(&qd)
-            .expect("queue descriptor should be in the queue table");
-
-        // Get reference to the underlying file descriptor.
-        // The following call to expect() is safe because when creating a queue we assigned it a valid file descritor.
-        let fd: RawFd = queue.get_fd().expect("queue should have a file descriptor");
-
-        // Create a socket that is bound to the target local address.
-        let bound_socket: Socket = {
-            let unbound_socket: &Socket = queue.get_socket();
-            unbound_socket.bind(local)?
-        };
-
-        // Bind underlying socket.
-        let saddr: SockAddr = linux::socketaddrv4_to_sockaddr(&local);
-        match unsafe { libc::bind(fd, &saddr as *const SockAddr, mem::size_of::<SockAddrIn>() as Socklen) } {
-            stats if stats == 0 => {
-                // Update socket.
-                queue.set_socket(&bound_socket);
+        // Issue bind operation.
+        let mut queue: CatnapQueue = self.get_queue(qd)?;
+        queue.prepare_bind()?;
```

Comment:
Should we go ahead and put this into the queue.bind() call?

Reply:
Yes, it could, but then the responsability for driving the state machine would be spread across the two layers. Eventually we want to place all of them in the queue layer, but this is not possible now.